src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/RabbitMq/HandleBasicDeliverWrapper.cs

```diff
@@ -70,5 +91,61 @@ IEnumerable<string> GetHeaderValue(IDictionary<string, object> carrier, string k
                 return null;
             }
         }
+
+        private void GetServerDetails(InstrumentedMethodCall instrumentedMethodCall, out string hostname, out int? port, IAgent agent)
+        {
+            if (_hasGetServerFailed)
+            {
+                hostname = null;
+                port = null;
+            }
+
+            try
+            {
+                _modelGetter ??= VisibilityBypasser.Instance.GeneratePropertyAccessor<object>(instrumentedMethodCall.MethodCall.InvocationTarget.GetType(), "Model");
+                var model = _modelGetter(instrumentedMethodCall.MethodCall.InvocationTarget);
+
+                object connection = null;
+                if (model.GetType().ToString() == "RabbitMQ.Client.Framing.Impl.Model")
```

Comment:
Would it be better to see if we can compare the type directly? For example you could have something like the following.
```
ConcurrentDictionary<Type, Action<Type>> _connectionGetter;

modelType = model.GetType()
connectionGetter = _connectionGetter.GetOrAdd(modelType, GetConnectionForType)

if (connectionGetter != null)
{
  connection = connectionGetter(modelType, model)
}

static Action<Type, Object> GetConnectionForType(Type modelType)
{
  if (modelType.ToString() == "framing model")
    return GetConnectionFromFramingModel;
  else if (modelType.ToString() == "autorecovery model")
    return GetConnectionFromAutorecoveryModel;
    
    return null;
}

static object GetConnectionFromFramingModel(Type modelType, object model)
{
   // The content currently in the if blocks for getting the specific connection for that type
}

Doing something like this can avoid the string comparisons every time this instrumentation runs (it would only be run the first time). You would probably need to run a benchmark to see if it's any faster though.
```

Reply:
I implemented something like the above and it does work.  Performance is about the same compared to string comparisons.  The initial cost is slightly higher, maybe 2000 ticks more for the above, but the second time its about 500 ticks less.  One advantage of this approach is that it is easier to add new instrumentation logic when new library versions are released.  So far we know that 4, 5, and 6 all need slightly different logic to get at the underlying data.  I am going to push up my commit and we can see what folks think.